include/deal.II/matrix_free/fe_evaluation.h

```diff
@@ -6719,7 +6719,7 @@ FEEvaluation<dim,
       message += "                                 " + correct_pos;
 
       Assert(static_cast<unsigned int>(fe_degree) ==
-                 this->data->data.front().fe_degree &&
+                 this->data->data.back().fe_degree &&
```

Comment:
I know this is a bit of an ugly fix... I wrote it so that `data.front().fe_degree = degree + 1` and `back() = degree`. One could rewrite it so that it is the same, but this also means a change of `+-1` in a few places in `include/deal.II/matrix_free/evaluation_kernels.h`. Do you think I should make this change? I guess I could also change the assert depending on if it is a RT element or not.

Reply:
I think the motivation for `FiniteElement::degree` is to give the highest degree in any coordinate direction, so that a quadrature formula with `FE::degree+1` ensures exact integration. I don't recall the precise convention, but I remember having discussed this with Guido Kanschat at some point (>10 years ago).